### DISCUSSION TITLE
ContainerID in Datadog

### DISCUSSION THREAD

**Discussion by damyan90** [open-telemetry/opentelemetry-java-instrumentation#4568] [Q&A] [answered]
Hi,

on the list of attributes sent to the Datadog with traces I see a `container.id` which I assume is used to automatically show also container metrics:
![image](https://user-images.githubusercontent.com/82033647/139869653-633dc4cb-d820-4c01-a7a2-6ffeb5698468.png)
![image](https://user-images.githubusercontent.com/82033647/139869722-52109a03-a699-45b8-bada-7facd251cc0f.png)

I'm curious how the `container.id` is retrieved from the k8s. It's for sure not exposed with the downwardAPI.

That would help me in configuration of a manual instrumented app with OTLP.

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @damyan90 ,
Please take a look at the [`ContainerResource`](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java#L62) class from the OpenTelemetry Java SDK - that's what the agent uses internally to compute container id.
